### PR TITLE
SDK-352 [Python SDK] Add ability to pass custom headers to Storage methods

### PR DESCRIPTION
**Pull request by RommeroInCountry**
Pull Request Checklist
======================

Please read this for our PR Culture:
------------------------------------
https://incountry.atlassian.net/wiki/spaces/ED/pages/46858869/Pull+requests+culture+Draft

Work relates to:
----------------
Link(s) to Jira Task/User Story/Bug/Epic:
- [SDK-352](https://incountry.atlassian.net/browse/SDK-352)

Code Quality (Before submitting PR)
------------
- [ ] Ensure code compiles and passes Sonar Lint.
- [ ] Do your best to ensure integration tests pass. If they don't then...
- [ ] Remove, ignore, or mark pending any brittle tests.
- [ ] Remove commented out code.

Target Release
--------------
- [ ] Please put the target release vehicle this needs to be bundled with.
- [ ] Did you increment the version? What is the new version?

### CODE REVIEW

File: tests/unit/storage/test_attachments.py [hunk 1]
```diff
@@ -342,11 +342,19 @@ def test_get_attachment_meta(client, response, encrypt):
...
 
-    res = client(encrypt).get_attachment_meta(country=COUNTRY, record_key=record_key, file_id=file_id)
+    res = client(encrypt).get_attachment_meta(
+        country=COUNTRY, record_key=record_key, file_id=file_id, request_options=request_options
+    )
 
     res.should.have.key("attachment_meta")
     assert res["attachment_meta"] == response
 
+    last_request = httpretty.last_request()
```
Comment (artemgolozubov) [inline tests/unit/storage/test_attachments.py:352]: Looks like this check is duplicated several times in this file. Please consider moving it to a separate function

File: tests/unit/storage/test_general.py [hunk 2]
```diff
@@ -45,8 +48,47 @@ def test_error_on_popapi_error(client, record, encrypt):
...
+            assert received_headers[k] == request_options["http_headers"][k]
+
+    client(encrypt).find_one.when.called_with(
+        country=COUNTRY, request_options=request_options, **record
+    ).should.have.raised(StorageServerException)
+    last_request = httpretty.last_request()
+    received_headers = last_request.headers
+    if request_options.get("http_headers"):
+        for k in request_options["http_headers"]:
+            assert received_headers[k] == request_options["http_headers"][k]
```
Comment (artemgolozubov) [inline tests/unit/storage/test_general.py:94]: The same here - duplicated code

**Review by artemgolozubov — APPROVED** [review]
LGTM

File: incountry/storage.py [hunk 3]
```diff
@@ -124,12 +126,17 @@ def __init__(
             ],
         }
     )
-    def write(self, country: str, record_key: str, **record_data: Union[str, int]) -> Dict[str, TRecord]:
+    def write(
+        self, country: str, record_key: str, request_options: dict = {}, **record_data: Union[str, int]
```
[the remark this answers is not in the document]
Comment (RommeroInCountry) [reply incountry/storage.py:130]: can't do that in python when having **argument :)

File: tests/unit/storage/test_general.py (hunk 2 above)
Comment (RommeroInCountry) [reply tests/unit/storage/test_general.py:94]: Probs I'll leave it as is now as. Moving it to separate function kinda breaks test case simplicity and I can't find an elegant solution for now

**Review by avdyakovs — APPROVED** [review]

**Pull request merged**